File: 리모컨.py
CUR_POS =100
MAX=500000
brokenNum = [False]* 10

n= int(input())
m= int(input())
if m > 0:
  a = list(map(int,input().split()))
else : 
  a = []
for x in a:
  brokenNum[x] = True;


def isPossible(x):
  if x == 0:
    if brokenNum[x]:
      return 0
    else:
      return 1
  cnt = 0
  while(x > 0):
    num = x % 10
    if brokenNum[num]:
      return 0
    cnt += 1
    x //= 10
  return cnt

ans = abs(CUR_POS-n)
for x in range(MAX*2 + 1):
  cnt = isPossible(x);
  if cnt == 0:
    continue
  cnt += abs(x-n)
  ans = min(ans,cnt)
print(ans)

File: 리모컨.cpp
#include <iostream>
#include <algorithm>

using namespace std;

const int CUR_POS=100;
const int MAX = 500000;
bool brokenNum[10];
//숫자 x를 버튼으로 누를 수 있는지 판단=>
// 누를수 없다면 0, 누를 수 있다면 X의 자릿수를 반환.
int isPossible(int x){
  //예외 처리 : x 가 0 
  if( x == 0 ){
    if(brokenNum[x]) return 0;
    else return 1;
  }
  // x가 1이상이라면
  int cnt=0;
  while( x != 0 ){
    int num = x%10;
    if(brokenNum[num]) return 0;
    else cnt+=1;
    x/=10;
  }
  return cnt;
}

int main(){
  int n,m; 
  cin>>n;
  cin>>m;
  while(m--){
    int btn;
    cin>>btn;
    brokenNum[btn]=true;
  }
  int ans = abs(CUR_POS-n);
  for(int x = 0; x<= MAX*2; x++){
    int cnt = isPossible(x);
    if(cnt == 0 ) continue;
    cnt += abs(x-n);
    ans= min(ans,cnt);
  }
  cout<<ans<<'\n';
  return 0;
}